src/structures/InteractionCollector.js

```diff
@@ -17,26 +23,49 @@ const { Events } = require('../util/Constants');
  * channel (`'channelDelete'`), or guild (`'guildDelete'`) are deleted.
  * @extends {Collector}
  */
-class MessageComponentInteractionCollector extends Collector {
+class InteractionCollector extends Collector {
   /**
-   * @param {Message|TextChannel|DMChannel|NewsChannel} source
-   * The source from which to collect message component interactions
-   * @param {MessageComponentInteractionCollectorOptions} [options={}] The options to apply to this collector
+   * @param {Client} client The client on which to collect message component interactions
+   * @param {InteractionCollectorOptions} [options={}] The options to apply to this collector
    */
-  constructor(source, options = {}) {
-    super(source.client, options);
+  constructor(client, options = {}) {
+    super(client, options);
 
     /**
-     * The message from which to collect message component interactions, if provided
+     * The message from which to collect interactions, if provided
      * @type {?Message}
      */
-    this.message = source instanceof require('./Message') ? source : null;
+    this.message = options.message ?? null;
 
     /**
-     * The source channel from which to collect message component interactions
-     * @type {TextChannel|DMChannel|NewsChannel}
+     * The channel from which to collect interactions, if provided
+     * @type {?TextChannel|DMChannel|NewsChannel}
```

Comment:
```suggestion
     * @type {?(TextChannel|DMChannel|NewsChannel)}
```

Reply:
isn't it the same type either way

Reply:
Doesn't TextBasedChannel work?

Reply:
TextBasedChannel isn't a jsdoc type, its an interface applied to other classes.